core/cog-viewport.c

```diff
@@ -60,6 +65,24 @@ G_DEFINE_TYPE_WITH_PRIVATE(CogViewport, cog_viewport, G_TYPE_OBJECT)
 
 #define PRIV(obj) ((CogViewportPrivate *) cog_viewport_get_instance_private((CogViewport *) obj))
 
+static void *
+_cog_viewport_get_impl_type_init(GType *type)
+{
+    *type = COG_TYPE_VIEWPORT;
+
+    CogPlatform *platform = cog_platform_get();
+    if (platform) {
```

Comment:
Nowadays we can safely assume that `cog_platform_get()` will always return a valid instance because we have `CogFallbackPlatform` always built into `libcogcore`.

I know this is copying what `_cog_view_get_impl_type_init()` is doing, so I think it's okay this way, but this is a reminder to make a follow-up patch later to simplify this 😉

Reply:
Split: https://github.com/Igalia/cog/issues/664